apm-agent-plugins/apm-es-restclient-plugin/apm-es-restclient-plugin-common/src/main/java/co/elastic/apm/agent/esrestclient/ResponseListenerWrapper.java

```diff
@@ -28,24 +30,69 @@
 public class ResponseListenerWrapper implements ResponseListener, Recyclable {
 
     private final ElasticsearchRestClientInstrumentationHelper helper;
+    private final Tracer tracer;
+
     @Nullable
     private ResponseListener delegate;
+
```

Comment:
[for reviewer] Here we can't easily sub-class due to the class instance pooling, thus there is effectively two implementations that are triggered depending on the use-case.